### ISSUE TITLE
[FreeBSD] Cannot allocate memory

### ISSUE THREAD

**Issue by avkarenow** [giampaolo/psutil#2032]
## Summary

* OS: FreeBSD 12.2/13.0
* Architecture: amd64
* Psutil version: 5.8.0
* Python version: 3.8.12
* Type: core

## Description
Hello,

From time to time while I'm running code like `for proc in psutil.process_iter(): ...` I got: Cannot allocate memory (originated from sysctl)
It happens on many FreeBSD 12.2 and 13.0 installations with 128-196 GB of RAM. Code is running as root with ulimit:
```
cpu time               (seconds, -t)  unlimited
file size           (512-blocks, -f)  unlimited
data seg size           (kbytes, -d)  33554432
stack size              (kbytes, -s)  524288
core file size      (512-blocks, -c)  unlimited
max memory size         (kbytes, -m)  unlimited
locked memory           (kbytes, -l)  unlimited
max user processes              (-u)  34318
open files                      (-n)  942921
virtual mem size        (kbytes, -v)  unlimited
swap limit              (kbytes, -w)  unlimited
socket buffer size       (bytes, -b)  unlimited
pseudo-terminals                (-p)  unlimited
kqueues                         (-k)  unlimited
umtx shared locks               (-o)  unlimited
```

```
Traceback:
    for proc in psutil.process_iter():
  File "/usr/local/lib/python3.8/site-packages/psutil/__init__.py", line 1425, in process_iter
    a = set(pids())
  File "/usr/local/lib/python3.8/site-packages/psutil/__init__.py", line 1365, in pids
    ret = sorted(_psplatform.pids())
  File "/usr/local/lib/python3.8/site-packages/psutil/_psbsd.py", line 513, in pids
    ret = cext.pids()
OSError: [Errno 12] Cannot allocate memory (originated from sysctl)
```

**Comment by peterjeremy** [comment]
This looks to be the same as #2093

**Comment by avkarenow** [comment]
I was testing last commit version (04e7aa604155736cce0abcc15c9b7b63d941b0e9) but the problem with cannot allocate memory still appears.

**Comment by avkarenow** [comment]
The issue seems to be fixed by c7c7bbcfc762d73a50d3cb2ad6f277dad7bb9a6b and works now without problems.
